test/Horoscope_spec.js

```diff
@@ -0,0 +1,64 @@
+var assert                        = require("assert");
+var helpers                       = require("./helpers");
+var calendar                      = require("../public/json/calendar.json");
+var grammar                       = require("../public/json/grammar.json");
+var sentenceBigramProbabilities   = require("../public/json/sentenceBigramProbabilities.json");
+var sentenceTypes                 = require("../public/json/sentenceTypes.json");
+var Horoscope                     = require("../public/js/Horoscope.js");
+
+describe("Horoscope form requirements", function(){
+    var horoscope = Horoscope.Horoscope;
+    console.log(horoscope);
+    before(function(){
+        validForm = new horoscope({userData: helpers.validForm});
+    });
+    describe("Form valid if...", function(){
+        it("all validators successful", function(){
+            assert(validForm.testing.validateForm(), "Not valid");
+        });
+    });
+    describe("Form invalid if...", function(){
+        it("hometown is blank", function(){
+            var invalidForm = new horoscope({userData : {hometown: ""}});
+            assert(!invalidForm.testing.hometownIsValid())
+        });
+        it("name is blank", function () {
+            var invalidForm = new horoscope({userData : {name : ""}});
+            assert(!invalidForm.testing.nameIsValid());
+        });
+        it("birthday is incomplete", function () {
+            var invalidForm = new horoscope({userData : {birthday : ""}});
+            assert(!invalidForm.testing.birthdayIsValid());
+        });
+    });
+});
+
+describe("Sign Evaluation", function(){
```

Comment:
These specs look really clean 😃.

I like how you have the `...` and `it` statements that complete the sentence. 

I'll just throw out my system in case it is of interest (I used to do it your way, but struggled to making the sentences line up and consistent structure).

I have a describe with the name of the module:

`describe('MyModule', function() { ... })`

Every static function on the module has a describe block. The description is `.<method name>`:

`describe('.doStuff', function () { ... })`

Every instance function on the module has a describe block. The description is `#<instance method name>`.

`describe('#instanceStuff', function () { ... })`

Test cases for that function are all in that describe:

```js
describe('MyModule', () => {
   describe('.doStuff', () => {
      it('does the thing', () => {
         assert(MyModule.doStuff() === "The Thing");
      });
   });

   describe('#instanceStuff', () => {
      it('does instance stuff', () => {
         var mod = new MyModule();
         assert(mod.instanceStuff() === "Instance Stuff");
      });
   });
});
```

Just makes it really easy to read through my tests and know exactly where failures are happening. But looks like these might be more integration style tests, which makes this system harder. 

But thought I'd share!